### ISSUE TITLE
'checkstyle_checks.xml' should check for proper array initialization RCURLY indentation

### ISSUE THREAD

**Issue by nmancus1** [checkstyle/checkstyle#9863]
Example:

```java

        final String[] expected = {
            "14:2: " + getCheckMessage(MSG_SEMI),
            "20:2: " + getCheckMessage(MSG_SEMI),
            }; // should be violation

```

Since IntelliJ does not enforce/ auto-format array initialization right curly brackets (unless indentation is less than specified) we could add the following modules to `checkstyle_checks.xml` to enforce this style:

```xml
        <module name="Indentation">
            <property name="id" value="arrayInitRCurly"/>
            <property name="forceStrictCondition" value="true"/>
            <property name="lineWrappingIndentation" value="0"/>
            <message key="indentation.error"
                     value="Array initialization right curly should be indented same level as parent declaration."/>
        </module>

        <module name="SuppressionXpathSingleFilter">
            <property name="checks" value="Indentation"/>
            <property name="id" value="arrayInitRCurly"/>
            <property name="query" value="//*[not(self::RCURLY and parent::ARRAY_INIT)]"/>
        </module>
```

Right now, this config produces 78 violations:
`/home/nick/IdeaProjects/checkstyle/config/ant-phase-verify.xml:174: Checkstyle failed: Got 78 errors and 0 warnings.`

Another way to achieve this would be to use XPATH to check if parent expression(VARIABLE_DEF) and RCURLY have the same indentation; but we would need to add column and line number attributes to ElementNode.
